enclave-runtime/src/rpc/author/author_container.rs

```diff
@@ -0,0 +1,94 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{
+	rpc::author::{
+		atomic_container::AtomicContainer, Author, AuthorApi, OnBlockCreated, SendState,
+	},
+	state::StateFacade,
+	top_pool::pool_types::BPool,
+};
+use sgx_crypto_helper::rsa3072::Rsa3072KeyPair;
+use sp_core::H256;
+use std::{
+	marker::PhantomData,
+	sync::{Arc, SgxMutex},
+};
+
+static GLOBAL_AUTHOR_CONTAINER: AtomicContainer = AtomicContainer::new();
+
+pub trait FullAuthor = AuthorApi<H256, H256>
+	+ SendState<Hash = H256>
+	+ OnBlockCreated<Hash = H256>
+	+ Send
+	+ Sync
+	+ 'static;
+
+/// getter trait for the RPC author
+pub trait GetAuthor: Send + Sync + 'static {
+	type AuthorType: FullAuthor;
+
+	fn get(&self) -> Option<&'static SgxMutex<Arc<Self::AuthorType>>>;
+}
+
+/// Global container wrapper for the RPC author
+/// must be initialized before use, calling the `initialize()` method
+pub struct GlobalAuthorContainer;
+
+impl GlobalAuthorContainer {
+	pub fn initialize(trusted_operation_pool: Arc<<Self as GetAuthor>::AuthorType>) {
+		GLOBAL_AUTHOR_CONTAINER.store(trusted_operation_pool)
+	}
+}
```

Comment:
We now have a global author container (replacing the global top pool container).